test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -183,34 +187,29 @@ database_operation::read_operation(thread_context *tc)
 {
     logger::log_msg(
       LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
-    WT_CURSOR *cursor = nullptr;
-    WT_DECL_RET;
+
     std::map<uint64_t, scoped_cursor> cursors;
     while (tc->running()) {
         /* Get a collection and find a cached cursor. */
         collection &coll = tc->db.get_random_collection();
-        const auto &it = cursors.find(coll.id);
-        if (it == cursors.end()) {
-            auto sc = tc->session.open_scoped_cursor(coll.name.c_str());
-            cursor = sc.get();
-            cursors.emplace(coll.id, std::move(sc));
-        } else
-            cursor = it->second.get();
+
+        if (cursors.find(coll.id) == cursors.end())
+            cursors.emplace(coll.id, std::move(tc->session.open_scoped_cursor(coll.name.c_str())));
+
+        /* Do a second lookup now that we know it exists. */
+        auto &it = cursors[coll.id];
```

Comment:
Can we call this something more descriptive like `cursor`?

Also to be precise, the type is not an iterator. An iterator is the return type of `find()`, `begin()`, `end()` or one of those functions.

Reply:
11c414a9d